common4j/src/main/com/microsoft/identity/common/java/util/UrlUtil.java

```diff
@@ -66,15 +76,28 @@ public static URL appendPathToURL(@NonNull final URL urlToAppend,
         final CommonURIBuilder builder = new CommonURIBuilder(urlToAppend.toString());
         final List<String> pathSegments = builder.getPathSegments();
 
-        final List<String> combinedPathSegments = new ArrayList<>(pathSegments);
+        final List<String> combinedPathSegments = new ArrayList<>();
 
+        // Add all non-empty path segments from the base URL
+        if (removeEmptySegments) {
+            for (final String path : pathSegments) {
+                if (!StringUtil.isNullOrEmpty(path)) {
+                    combinedPathSegments.add(path);
+                }
+            }
+        }
```

Comment:
I'm not exactly sure I understand what's going on here and why do you need this code. Perhaps this is something where you shouldn't have modified existing code and rather add a new method but seems you've modified existing code in this case